### PR TITLE
fix: enforcing content-type header while registering the instance id

### LINKED ISSUE #20949: [Bug]:  CS logs are filled with exceptions related to server start-up
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Cloud services logs are filled with exceptions which happens due to instanceId not being found, when server queries for remote plugins.

### Steps To Reproduce

To reproduce, try to hit these apis

- {{cloudServices}}/api/v1/installations with object {"key" : "id"}, make content type as application/octet-stream in request header.
-  {{cloudServices}}/api/v1/plugins?instanceId={{id}}, (this throws unauthorized access)

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

self hosted

### PR DESCRIPTION
**Pull request by sondermanish**
## Description
In cloud services logs we are experiencing tons of `415 application/octet-stream `exceptions  related to the `api/v1/installation`. i.e when we register a new instance from the cloud services. as a consequence instance ids can't be registered.
This might be happening due to automatic setting of content-type to octet-stream, can't say for sure.
Here we are enforcing a "application/json" content-type.

- We are also experiencing unauthorised access exception logs thrown when we try to get remote plugins for the instance 
on endpoint `api/v1/plugins?instanceId={{instanceId}}` 
According to our hypothesis this one is thrown because the instance couldn't be registered in first place. hence the repo has no history of this particular instance id

> TL;DR enforced a application/json content type on api call for registering instance.

[Slack Thread](https://theappsmith.slack.com/archives/CGBPVEJ5C/p1677074154818309)

Fixes #20949 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
This has been tested:
- Manual

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Review by subrata71 — APPROVED** [review]
LGTM

**Comment by sondermanish** [comment]
ok-to-test sha=34d3899

**Comment by sondermanish** [comment]
As this change is related to instance registration, it doesn't come under QA purview.

**Comment by sondermanish** [comment]
perf test failing on this one.
